### PR TITLE
Patches for catalog-to-dd

### PR DESCRIPTION
**Pull request by calum-chamberlain**
## What does this PR do?

1. Check only the first char of the phase hint
2. Warn about different lengths being found, which occurs due to insufficient data passed
3. Skip correlations when no matching data are found.

### Why was it initiated?  Any relevant Issues?

@emilyws1 ran into more issues running catalog to dd when data were not trimmed correctly and phases had "odd" phase hints.

### PR Checklist
- [x] `develop` base branch selected?
- [x] This PR is not directly related to an existing issue (which has no PR yet).
- [x] All tests still pass.
- [x] Any new features or fixed regressions are be covered via new tests.
- [x] Any new or changed features have are fully documented.
- [x] Significant changes have been added to `CHANGES.md`.
~~- [ ] First time contributors have added your name to `CONTRIBUTORS.md`.~~

### CODE REVIEW

File: eqcorrscan/utils/catalog_to_dd.py [hunk 1]
```diff
@@ -262,19 +262,35 @@ def _compute_dt_correlations(catalog, master, min_link, event_id_mapper,
...
+            if len(set(master_length)) > 1:
+                Logger.warning("Multiple lengths found - check that you "
+                               "are providing sufficient data")
+            master_length = Counter(master_length).most_common(1)[0][0]
             _master_stream = _master_stream.select(npts=master_length)
-            matched_length = Counter(
-                (tr.stats.npts for st in _matched_streams.values()
-                 for tr in st)).most_common(1)[0][0]
+            
+            matched_length = [tr.stats.npts for st in _matched_streams.values() for tr in st]
```
Comment (calum-chamberlain) [inline eqcorrscan/utils/catalog_to_dd.py:272]:
```suggestion
            matched_length = [
                tr.stats.npts for st in _matched_streams.values() for tr in st]
```

File: eqcorrscan/utils/catalog_to_dd.py [hunk 2]
```diff
@@ -262,19 +262,35 @@ def _compute_dt_correlations(catalog, master, min_link, event_id_mapper,
...
             # Remove empty streams and generate an ordered list of event_ids
             used_event_ids, used_matched_streams = [], []
             for event_id, _matched_stream in _matched_streams.items():
                 _matched_stream = _matched_stream.select(npts=matched_length)
                 if len(_matched_stream) > 0:
                     used_event_ids.append(event_id)
                     used_matched_streams.append(_matched_stream)
+            # Check that there are matching seed ids.
+            master_seed_ids = set(tr.id for tr in _master_stream)
+            matched_seed_ids = set(tr.id for st in used_matched_streams for tr in st)
```
Comment (calum-chamberlain) [inline eqcorrscan/utils/catalog_to_dd.py:287]:
```suggestion
            matched_seed_ids = set(
                tr.id for st in used_matched_streams for tr in st)
```

File: eqcorrscan/utils/catalog_to_dd.py [hunk 3]
```diff
@@ -311,7 +327,7 @@ def _compute_dt_correlations(catalog, master, min_link, event_id_mapper,
                         _DTObs(station=chan.channel[0],
                                tt1=master_tts["{0}_{1}".format(
                                    chan.channel[0], phase_hint)],
-                               tt2=tt2, weight=cc_max ** 2, phase=phase_hint))
+                               tt2=tt2, weight=cc_max ** 2, phase=phase_hint[0]))
```
Comment (calum-chamberlain) [inline eqcorrscan/utils/catalog_to_dd.py:330]:
```suggestion
                               tt2=tt2, weight=cc_max ** 2, 
                               phase=phase_hint[0]))
```

File: eqcorrscan/utils/catalog_to_dd.py [hunk 4]
```diff
@@ -262,19 +262,38 @@ def _compute_dt_correlations(catalog, master, min_link, event_id_mapper,
...
+                Logger.warning("Multiple lengths found - check that you "
+                               "are providing sufficient data")
+            master_length = Counter(master_length).most_common(1)[0][0]
             _master_stream = _master_stream.select(npts=master_length)
-            matched_length = Counter(
-                (tr.stats.npts for st in _matched_streams.values()
-                 for tr in st)).most_common(1)[0][0]
+            
+            matched_length = [
+                tr.stats.npts for st in _matched_streams.values()
```
Comment (calum-chamberlain) [inline eqcorrscan/utils/catalog_to_dd.py:273]:
```suggestion
                tr.stats.npts for st in _matched_streams.values()
```

File: eqcorrscan/utils/catalog_to_dd.py [hunk 5]
```diff
@@ -311,7 +330,8 @@ def _compute_dt_correlations(catalog, master, min_link, event_id_mapper,
                         _DTObs(station=chan.channel[0],
                                tt1=master_tts["{0}_{1}".format(
                                    chan.channel[0], phase_hint)],
-                               tt2=tt2, weight=cc_max ** 2, phase=phase_hint))
+                               tt2=tt2, weight=cc_max ** 2,
```
Comment (calum-chamberlain) [inline eqcorrscan/utils/catalog_to_dd.py:333]:
```suggestion
                               tt2=tt2, weight=cc_max ** 2,
```

**Comment by cjhopp** [comment]
@calum-chamberlain Maybe better as a separate feature request but how do you feel about adding station elevations to `write_station()`?

I'm not sure if this breaks backward compatibility for HypoDD <2.1b, so maybe is best left to the user to write their own func.

**Comment by calum-chamberlain** [comment]
Adding elevations would be simple and worthwhile.  Adding it as an option would solve compatibility. Good idea.  Do you have a suggested edit? I had forgotten that I could use elevations in 2.1!

**Comment by cjhopp** [comment]
Does this commit look okay? I also added a change to fix the following:

(Typed this an hour ago and forgot to comment...) 

I actually just ran into the `len(phase_hint) > 1` issue so I've been testing this branch.

I don't think [these lines](https://github.com/eqcorrscan/EQcorrscan/blob/38cf07e7bceebf0e12217ad61f41f34b01d7ed7c/eqcorrscan/utils/catalog_to_dd.py#L287-L295) are doing what they're supposed to. They return things like this: 
`After checking length there are no matched traces: master: {'UW.HART..HNZ', 'UW.RVC..EHZ'}, matched: {'UW.HART..HNZ', 'UW.RVC..EHZ'}`.

Maybe we want something like: `if matched_seed_ids.issubset(master_stream_ids)`?

**Comment by calum-chamberlain** [comment]
That is what I meant.  Darn. Need to add a test to this pr...

**Comment by calum-chamberlain** [comment]
NCEDC is timing out at the moment, but it looks like everything else is passing. I fixed the formatting for `write_station` to use elevations and tested with `ph2dt` to confirm that this works. I also updated to use higher precision lat and lon. 

I will try re-running the tests in the morning and merge if the relevant ones pass - fingers-crossed NCEDC is running faster tomorrow!

**Comment by calum-chamberlain** [comment]
NCEDC is still timing out. Merging.

**Pull request merged**